nav2_simple_commander/nav2_simple_commander/footprint_collision_checker.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python3
+# Copyright 2021 Samsung Research America
+# Copyright 2022 Afif Swaidan
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from math import cos, sin
+from line_iterator import LineIterator
+from costmap_2d import PyCostmap2D
+from geometry_msgs.msg import Point
+
+NO_INFORMATION = 255
+LETHAL_OBSTACLE = 254
+INSCRIBED_INFLATED_OBSTACLE = 253
+MAX_NON_OBSTACLE = 252
+FREE_SPACE = 0
+
+
+class FootprintCollisionChecker():
+    def __init__(self, costmap: PyCostmap2D):
+        self.costmap_ = costmap
+
+    def footprintCost(self, footprint: list[Point]):
```

Comment:
a doc string would be nice here to explain the units: Is the footprint in world or map coords?

Reply:
Alright noted,  I wanted to add the docstrings and the unit tests after I take your initial opinion.